src/client/screens/settings/Settings.js

```diff
@@ -43,35 +52,155 @@ export default class Settings extends React.Component {
+          await cleanUp()
+        } else
+        // is not registered? then register
+        if (!this.state.registered && this.state.notifications) {
+          await this._handlePushRegistration()
+          this.setState({ registered: true })
+          await setRegisterd(true)
+          await cleanUp()
+        } else
+        //  we are registerd so update instead
+        if (this.state.registered && this.state.notifications) {
+          await this._handlePushRegistrationUpdate()
+          await cleanUp()
+        } else {
+          console.info('nothing to do...')
+          await cleanUp()
+        }
+      } catch (err) {
+        console.info(err)
+      }
+    }
+
+    const cleanUp = () => {
+      console.info('sync was successfull. cleaning up.')
+
+      // remove this job
+      clearInterval(this.syncJob)
+      this.syncJob = null
+
+      // we are officially synced
+      this.setState({ synced: true })
+      return setSynced(true)
+    }
+
+    this.syncJob = setInterval(job, 30000)
+    job()
+  }, 3000)
+
+  _handlePushRegistrationUpdate = async () => {
+    let token = await Notifications.getExpoPushTokenAsync()
+    return fetch(`${env.API_URL}/register/${token}`, {
+      method: 'PUT',
+      headers: {
+        Accept: 'application/json', 'Content-Type': 'application/json'
+      },
+      body: JSON.stringify({
+        ...this.state,
+        token: token
+      })
+    })
+  }
+
+  _handlePushRegistration = async (register) => {
+    // Get the token that uniquely identifies this device
+    let token = await Notifications.getExpoPushTokenAsync()
+    return fetch(`${env.API_URL}/register`, {
+      method: 'POST',
+      headers: {
+        Accept: 'application/json', 'Content-Type': 'application/json'
+      },
+      body: JSON.stringify({
+        ...this.state,
+        token: token
+      })
+    })
+  }
+
+  _handlePushDeRegistration = async (register) => {
+    // Get the token that uniquely identifies this device
+    let token = await Notifications.getExpoPushTokenAsync()
+    return fetch(`${env.API_URL}/deregister/${token}`, {
+      method: 'DELETE'
+    })
   }
 
   _handleNotificationToggle = async value => {
-    this.setState({ notifications: value })
+    if (value) {
+      const { status: existingStatus } = await Permissions.getAsync(
+        Permissions.NOTIFICATIONS
+      )
+      let finalStatus = existingStatus
+      // only ask if permissions have not already been determined, because
+      // iOS won't necessarily prompt the user a second time.
+      if (existingStatus !== 'granted') {
+        // Android remote notification permissions are granted during the app
+        // install, so this will only ask on iOS
+        const { status } = await Permissions.askAsync(Permissions.NOTIFICATIONS)
+        finalStatus = status
+      }
+      // Stop here if the user did not grant permissions
+      if (finalStatus !== 'granted') {
+        return
```

Comment:
Hier gibts nen kleines Problem.. Hat man die Permissions nicht, springt man komplett aus der Funktion raus und aktualisiert den state (`this.state.notifications`) nicht. Wenn man sich anstatt dem `return` eine Flag setzt (bspw. `permissions`), wird in der sync Funktion anschließend trotzdem versucht eine Push zu senden. Da müsste man noch einmal drauf eingehen.